File: PARK_HYEONG_JUN/src/feb/_0205/Boj_1021.java
package feb._0205;

import java.util.ArrayDeque;
import java.util.Deque;
import java.util.Scanner;

public class Boj_1021 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();

        Deque<Integer> deque = new ArrayDeque<>();
        for (int i = 1; i <= n; i++) {
            deque.add(i);
        }

        int current = deque.peekFirst();
        int result = 0;
        for (int i = 0; i < m; i++) {
            int number = sc.nextInt();
            int count = 0;
            int value = 0;
            while ((value = deque.pollFirst()) != number) {
                count++;
                deque.add(value);
            }

            if (count > deque.size() + 1 - count) {
                count = deque.size() + 1 - count;
            }

            result += count;
        }

        System.out.println(result);
    }
}

File: PARK_HYEONG_JUN/src/feb/_0205/Boj_1021.kt
package feb._0205

import java.util.Scanner

fun main() {
    val reader = Scanner(System.`in`)
    val n = reader.nextInt()
    val m = reader.nextInt()

    val deque = ArrayDeque<Int>()

    for (i in 1..n) {
        deque.add(i)
    }

    var result = 0
    for (i in 0 until m) {
        val num = reader.nextInt()
        var count = 0

        while (deque.isNotEmpty()) {
            val value = deque.removeFirst()
            if (value == num) break

            count++
            deque.add(value)
        }

        if (count > deque.size + 1 - count) {
            count = deque.size + 1 - count
        }

        result += count
    }

    println(result)
}